src/utils/type_list.hpp

```diff
@@ -0,0 +1,91 @@
+//========================================================================================
+// (C) (or copyright) 2020-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#ifndef UTILS_TYPE_LIST_HPP_
+#define UTILS_TYPE_LIST_HPP_
+
+#include <string>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace parthenon {
+template <class... Args>
+struct TypeList {
```

Comment:
Would you mind adding a minimum amount of documentation, like two lines here what TypeList are about and two lines in the general developer doc highlighting that they are available and their use case?

Reply:
done